pkg/tests/ossm/smoke_test.go

```diff
@@ -117,6 +107,26 @@ func TestSmoke(t *testing.T) {
 			})
 		})
 
+		t.NewSubTest(fmt.Sprintf("install smcp %s", toVersion)).Run(func(t TestHelper) {
+			t.Logf("This test checks whether SMCP %s deletion deletes all the resources", env.GetSMCPVersion())
+
+			t.LogStepf("Install SMCP %s", toVersion)
+			assertSMCPDeploysAndIsReady(t, toVersion)
+			assertRoutesExist(t)
+
+			t.LogStep("Install bookinfo pods and sleep pod")
+			app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+
+			t.LogStep("Check if bookinfo traffic flows through the Proxy")
+			assertTrafficFlowsThroughProxy(t, ns)
+
+			t.LogStep("verify proxy startup time. Expected to be less than 10 seconds")
+			t.Log("Jira related: https://issues.redhat.com/browse/OSSM-3586")
+			t.Log("From proxy json , verify the time between status.containerStatuses.state.running.startedAt and status.conditions[type=Ready].lastTransitionTime")
+			t.Log("The proxy startup time should be less than 10 seconds for ratings pod")
+			assertProxiesReadyInLessThan10Seconds(t, ns)
```

Comment:
Given that we want to run the exact same checks after a clean install as well as after an upgrade, all these checks should be extracted into a function and called from both subtests. This way, you clearly show the intent / semantics. Also, if someone adds additional checks in the future, they will add them to the function, and thus to both subtests. Without the function, there's a high chance that they'll forget to add the check to one of the subtests.

Reply:
Ok